### DISCUSSION TITLE
Command line usage of DiaNN in HPC cluster

### DISCUSSION THREAD

**Discussion by 21maan** [vdemichev/DiaNN#799] [Q&A]
Hello
I am new to DiaNN and am trying to set it up on a HPC cluster. I tried using the following command prompt on your Github page, but got the message 'command not found'. Has anyone set up the default DiaNN software on HPC? If you have related Linux code, would be great if you could share with me. Thanks!
diann.exe --cfg C:\Users\21maa\Desktop\d27iul23\260723.d

**Comment by vdemichev** [comment]
To run .exe under Linux, will need to do it under Wine 6.8 or later.
Alternatively, just use the native Linux binaries. 
Great example - quantms (not to be confused with QuantUMS :))

**Comment by 21maan** [comment]
Sorry but I am not a coder. Can you elaborate on native Linus commands? I also cannot find a code when I searched for 'quantms'. So I am using MobaXTerm for running Linux commands. Is it possible to get the code to 1. load a .d file, 2. load a FASTA file and 3. specify input and output directories? I want to be able to run one .d file at a time, using default settings. If you can provide a link to where this code might be available that would be great. I am attaching screenshots of how this would ideally look like on your interface, the pages where I got the command line prompts from, and the error I am getting on my end when I follow these prompts on MobaXTerm (I am initially running it on my home directory, and have DiaNN installed on my laptop). Hoping you could help, thanks heaps! 
![image](https://github.com/vdemichev/DiaNN/assets/144721916/d530f2f4-acd8-4e1d-9c94-1252a36b5faf)
![image](https://github.com/vdemichev/DiaNN/assets/144721916/f85ea64d-dcec-4cb2-a2ed-a5be50f829da)
![image](https://github.com/vdemichev/DiaNN/assets/144721916/53b9f694-0fcd-4ff1-93ef-c0dfce1888f1)
![image](https://github.com/vdemichev/DiaNN/assets/144721916/8656c7d1-4d93-4e21-b9d4-22b882945c74)

**Reply by 21maan** [reply]
Hello just checking to see if  someone can help with my query above? Thanks

**Reply by vdemichev** [reply]
diann.exe command not found - means diann.exe simply not being launched, does not matter here whether --cfg or some other option is being passed to it. Have you tried the latest Wine?

**Reply by MichelMoser** [reply]
Hi @21maan , 

To run DIA-NN on HPC cluster you should use the DIA-NN docker container. You find it here: https://hub.docker.com/r/biocontainers/diann/tags
Get yourself familiar with the HPC job scheduler (probably SLURM) and apptainer or docker to run DIA-NN from the command line. 

Herer an example: 

```
DATA_PATH=/path/to/your/files
TOFFILE1=your_tof_file.d

OUT_REPORT=report.tsv
OUT_LIB=lib.tsv
PROT_LIB=protein_db.fasta

docker run -v ${DATA_PATH}:/data/ biocontainers/diann:v1.8.1_cv1 diann \
--f $TOFFILE1 \
--lib "" \
--threads 32 \
--verbose 2 \
--out $OUT_REPORT \
--qvalue 0.05 \
--matrix-spec-q 0.02 \
--matrices \
--min-corr 2.0 \
--corr-diff 1.0 \
--time-corr-only \
--extracted-ms1 \
--out-lib $OUT_LIB \
--gen-spec-lib \
--predictor \
--fasta $PROT_LIB \
--fasta-search \
--min-fr-mz 200 \
--max-fr-mz 1800 \
--met-excision \
--cut K*,R* \
--missed-cleavages 3 \
--min-pep-len 5 \
--max-pep-len 32 \
--min-pr-mz 200 \
--max-pr-mz 1800 \
--min-pr-charge 2 \
--max-pr-charge 4 \
--unimod4 \
--var-mods 2 \
--var-mod UniMod:35,15.994915,M \
--var-mod UniMod:1,42.010565,*n \
--monitor-mod UniMod:1 \
--mass-acc 10 \
--mass-acc-ms1 20 \
--reanalyse \
--relaxed-prot-inf \
--smart-profiling \
--peak-center \
--no-ifs-removal 

```

Adjust options to your needs! 

Cheers
